rfcs/text/refactoring-date-component.md

```diff
@@ -0,0 +1,330 @@
+- Start Date: (2021-07-28)
+
+# Table of contents
+
+- [Summary](#summary)
+- [Basic example](#basic-example)
+- [Motivation](#motivation)
+- [Detailed design](#detailed-design)
+  - [Replacing Momentjs](#replacing-momentjs)
+  - [Continuing to use React DayPicker](#continuing-to-use-react-daypicker)
+  - [Locales](#locales)
+  - [Event handling](#event-handling)
+  - [Support for hidden input](#support-for-hidden-input)
+  - [DatePicker component](#datepicker-component)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+- [Unresolved questions](#unresolved-questions)
+
+# Summary
+
+The purpose of this document is to outline a new simplified `Date` component implementation, removing any technical debt or logic that is specifically tied to any consuming project whilst supporting consumers in being able to implement their requirements in their own codebases.
+
+# Basic example
+
+The new implementation will be written as a functional component and only support being used as a controlled input: it will require the implementing developer to provide `value` and `onChange` props. The `onBlur` prop will also be required so the component can handle reformatting the value string based on the locale supplied via an `I18nProvider`. This will allow implementation teams to implement the component as follows:
+
+```jsx
+  // my-component.component.js
+  const MyComponent = () => {
+    const [value, setValue] = useState("28/07/1987");
+
+    const updateDateValue = (ev) => {
+      const { formattedValue } = ev.target.value;
+      setVisibleValue(formattedValue);
+    };
+
+    return (
+      <I18nProvider>
```

Comment:
By default carbon uses `en-GB` translations so you don't need to wrap it in `I18nProvider` for default.
https://carbon.sage.com/?path=/story/documentation-i18n--page#default

Reply:
Yeah this is just for the sake of being readable and making it clear how things would work logically, I'll probably expand the example to include a custom locale object once the first round of reviews are done

Reply:
Decided to remove it as it's redundant in this example